docs/dg/dev/troubleshooting/troubleshooting-general-technical-issues/session-locking-issues.md

```diff
@@ -0,0 +1,177 @@
+
+### Part B: Advanced session handling for read-only requests
+
+After implementing async forms, your GET requests are now truly read-only. You can instruct the system to use a more performant session handler that only locks on write operations.
+
+Create a custom session handler resolver that selects a handler based on the HTTP request method. In `\Spryker\Yves\SessionRedis\Resolver\SessionHandlerResolver::resolveConfigurableRedisLockingSessionHandler` configure `\Spryker\Shared\SessionRedis\Handler\SessionHandlerFactory::createSessionHandlerRedisWriteOnlyLocking()` to be used for GET requests. Example:
+
+**Pyz\Yves\SessionRedis\Resolver\SessionHandlerResolver.php**
+
+```php
+<?php
+
+namespace Pyz\Yves\SessionRedis\Resolver;
+
+use Spryker\Shared\SessionRedis\Handler\SessionHandlerFactory;
+use Spryker\Shared\SessionRedis\Redis\SessionRedisWrapperInterface;
+use Spryker\Yves\SessionRedis\Resolver\SessionHandlerResolver as SprykerSessionHandlerResolver;
+use Symfony\Component\HttpFoundation\Request;
+
+class SessionHandlerResolver extends SprykerSessionHandlerResolver
+{
+    /**
+     * @param \Spryker\Shared\SessionRedis\Redis\SessionRedisWrapperInterface $sessionRedisWrapper
+     *
+     * @return \SessionHandlerInterface
+     */
+    protected function resolveConfigurableRedisLockingSessionHandler(SessionRedisWrapperInterface $sessionRedisWrapper): \SessionHandlerInterface
+    {
+        // For truly read-only GET requests, use a handler that only locks on writes.
+        if ($this->requestStack->getCurrentRequest()->isMethod(Request::METHOD_GET)) {
+            return $this->sessionHandlerFactory->createSessionHandlerRedisWriteOnlyLocking($sessionRedisWrapper);
+        }
+
+        // For all other requests (POST, etc.), use the default locking handler.
+        return parent::resolveConfigurableRedisLockingSessionHandler($sessionRedisWrapper);
+    }
+}
+```
+
+### Combined result
+
+| User Type   | Request Type     | Description                                                                                                         | Locking Behavior                 |
+|-------------|------------------|---------------------------------------------------------------------------------------------------------------------|----------------------------------|
+| Bot         | GET              | Crawlers hit async pages; uses WriteOnlyLocking handler. No session write, so no Redis lock is created.             | No lock                          |
+| Real user   | GET              | Initial page load is fast and does not involve locking.                                                             | No lock                   |
+| Real user   | POST / AJAX      | AJAX calls to fetch forms or submit data use the default fully-locking handler to ensure data integrity.            | Fully-locking (default handler)  |
```

Comment:
is there a non-full lock?

Reply:
There is a write only locker which is described on this page.